### PR TITLE
Added Custom Field field.json snippets

### PR DESCRIPTION
**Pull request by beacondigitalmarketing**
Added all available fields as well as groups and repeater.

### CODE REVIEW

File: snippets/man_gen/hubl_custom_module_fields.json [hunk 1]
```diff
@@ -0,0 +1,519 @@
...
+      
+    ],
+    "description": "Repeater Group"
+  },
+  "Date Field": {
+    "prefix": "field.date",
+    "body": [
+      "{",
+      "\"id\": \" \",",
+      "\"name\": \"${1:date_field}\",",
```
Comment (williamspiro) [inline snippets/man_gen/hubl_custom_module_fields.json]: Do you think it is worth having placeholders for all of the names/labels as you do here?

File: snippets/man_gen/hubl_custom_module_fields.json [hunk 2]
```diff
@@ -0,0 +1,519 @@
...
+      "\"default\" : {",
+      "\"url\" : {",
+      "\"content_id\" : null,",
+      "\"type\" : \"EXTERNAL\",",
+      "\"href\" : \"\" },",
+      "\"open_in_new_tab\" : false,",
+      "\"no_follow\" : false",
+      "} }"
+    ],
+    "description": "Link Field"
```
Comment (williamspiro) [inline snippets/man_gen/hubl_custom_module_fields.json]: I wonder if all of the descriptions should be formatted as:
HubSpot Custom Module: Link Field
etc. with the "HubSpot Custom Module" prefix.

Thoughts?

File: snippets/man_gen/hubl_custom_module_fields.json [hunk 3]
```diff
@@ -0,0 +1,519 @@
...
+        "\"label\": \"Blog field\",",
+        "\"sortable\": false,",
+        "\"required\": false,",
+        "\"locked\": false,",
+        "\"hierarchical\": false,",
+        "\"max_depth\": 0,",
+        "\"type\": \"blog\",",
+        "\"default\": null }"
+      ],
+      "description": "blog Field"
```
Comment (williamspiro) [inline snippets/man_gen/hubl_custom_module_fields.json]: Capital B :)

**Comment by williamspiro** [comment]
Something else I just noticed, do we want the JSON output to be indented? 
Ex:
```
{
    "id" : "",
    "name": "email_field",
    "label": "Email address field",
    "sortable": false,
    "required": false,
    "locked": false,
    "hierarchical": false,
    "max_depth": 0,
    "type": "email",
    "default": null
}
```

rather than:
```
{
"id" : "",
"name": "email_field",
"label": "Email address field",
"sortable": false,
"required": false,
"locked": false,
"hierarchical": false,
"max_depth": 0,
"type": "email",
"default": null }
```
thoughts?

File: snippets/man_gen/hubl_custom_module_fields.json [hunk 4]
```diff
@@ -0,0 +1,519 @@
...
+      "\"required\" : false,",
+      "\"locked\" : false,",
+      "\"hierarchical\" : false,",
+      "\"max_depth\" : 0,",
+      "\"supported_types\" : [ \"EXTERNAL\", \"CONTENT\", \"FILE\", \"EMAIL_ADDRESS\", \"BLOG\" ],",
+      "\"type\" : \"link\",",
+      "\"default\" : {",
+      "\"url\" : {",
+      "\"content_id\" : null,",
+      "\"type\" : \"EXTERNAL\",",
```
Comment (williamspiro) [inline snippets/man_gen/hubl_custom_module_fields.json]: Might be cool if this was something like `${1|EXTERNAL,CONTENT,FILE,EMAIL_ADDRESS,BLOG|}` so the developer could easily choose a supported type

File: snippets/man_gen/hubl_custom_module_fields.json (hunk 1 above)
Comment (beacondigitalmarketing) [reply snippets/man_gen/hubl_custom_module_fields.json]: Personally, yes.  For fools like me, it is helpful to see this.  I can look at the design manager and see the labels and know what did what in the code. I do think it worth adding the cursor placement to all.  you will note that i started on the first, just didn't make it beyond that :)

File: snippets/man_gen/hubl_custom_module_fields.json (hunk 2 above)
Comment (beacondigitalmarketing) [reply snippets/man_gen/hubl_custom_module_fields.json]: makes sense.  can amend.  This being my first time making a vscode snippet, i am unclear where the description is represented for the user.  If this is something that is visible, i would love to add some helpers here.  eg showing the tag parameters.  anything that helps the coder

File: snippets/man_gen/hubl_custom_module_fields.json (hunk 3 above)
Comment (beacondigitalmarketing) [reply snippets/man_gen/hubl_custom_module_fields.json]: will fix :)

File: snippets/man_gen/hubl_custom_module_fields.json (hunk 4 above)
Comment (beacondigitalmarketing) [reply snippets/man_gen/hubl_custom_module_fields.json]: YES!

**Comment by beacondigitalmarketing** [comment]
> Something else I just noticed, do we want the JSON output to be indented?
> Ex:
> 
> ```
> {
>     "id" : "",
>     "name": "email_field",
>     "label": "Email address field",
>     "sortable": false,
>     "required": false,
>     "locked": false,
>     "hierarchical": false,
>     "max_depth": 0,
>     "type": "email",
>     "default": null
> }
> ```
> 
> rather than:
> 
> ```
> {
> "id" : "",
> "name": "email_field",
> "label": "Email address field",
> "sortable": false,
> "required": false,
> "locked": false,
> "hierarchical": false,
> "max_depth": 0,
> "type": "email",
> "default": null }
> ```
> 
> thoughts?

I am totally fine with making that update.  sadly, prettier did not like working with my content

File: snippets/man_gen/hubl_custom_module_fields.json (hunk 2 above)
Comment (williamspiro) [reply snippets/man_gen/hubl_custom_module_fields.json]: field

Here is a gif of how it all plays out (my indentation is still messed up 😆 )! You can preview your local changes by clicking F5 and opening up a .json file. The description shows up in the intellisesne window! And you can see how the `${1|EXTERNAL,CONTENT,FILE,EMAIL_ADDRESS,BLOG|}` prompts the selection after initiating the snippet! 😄

**Comment by beacondigitalmarketing** [comment]
updated branch with all requests.  tried to clean up indents generally as well.  would like to add more to the description in a future iteration.

**Comment by williamspiro** [comment]
HAHHA "intenion" :onion:

@beacondigitalmarketing let me know when this is good to merge and deploy 🚀💯

**Comment by beacondigitalmarketing** [comment]
All good on my end!

**Comment by williamspiro** [comment]
Just deployed version `0.1.4`! Thanks @beacondigitalmarketing !!!! 🚀

**Pull request merged**